src/role.ts

```diff
@@ -0,0 +1,380 @@
+   * (required).
+   */
+  readonly namespace: string;
+}
+
+/**
+ * Role is a namespaced, logical grouping of PolicyRules that can be referenced
+ * as a unit by a RoleBinding.
+ */
+export class Role extends RoleBase implements IRole {
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  public readonly resourceType = 'roles';
+
+  constructor(scope: Construct, id: string, props: RoleProps) {
+    super(scope, id);
+
+    if (props.metadata?.namespace && props.metadata.namespace !== props.namespace) {
+      throw new Error('If `metadata.namespace` is passed as an option, its value must match the value of `namespace`.');
+    }
+
+    const metadata: ApiObjectMetadata = {
+      ...props.metadata,
+      namespace: props.namespace,
+    };
+
+    this.apiObject = new k8s.KubeRole(this, 'Resource', {
+      metadata,
+      rules: Lazy.any({ produce: () => this.synthesizeRules() }),
+    });
+  }
+
+  private synthesizeRules(): any {
+    return undefinedIfEmpty(this._rules.map(rule => rule._toKube()));
+  }
+
+  /**
+   * Create a RoleBinding that binds the permissions in this Role
+   * to a list of subjects, that will only apply this role's namespace.
+   * @param subjects a list of subjects to bind to
+   */
+  public bind(...subjects: ISubject[]): RoleBinding {
+    const binding = new RoleBinding(this, 'RoleBinding', {
+      namespace: this.metadata.namespace!,
+      role: this,
+    });
+    binding.addSubjects(...subjects);
+    return binding;
+  }
+}
+
+/**
+ * Represents a cluster-level role.
+ */
+export interface IClusterRole extends IResource {
+
+}
+
+/**
+ * Options for `ClusterRole`.
+ */
+export interface ClusterRoleProps extends RoleCommonProps {
+  /**
+   * Specify labels that should be used to locate ClusterRoles, whose rules
+   * will be automatically filled into this ClusterRole's rules.
+   */
+  readonly aggregationLabels?: { [key: string]: string };
+}
+
+/**
+ * ClusterRole is a cluster level, logical grouping of PolicyRules that can be
+ * referenced as a unit by a RoleBinding or ClusterRoleBinding.
+ */
+export class ClusterRole extends RoleBase implements IClusterRole {
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  public readonly resourceType = 'clusterroles';
+
+  private readonly _labelSelector: Record<string, string>;
+
+  constructor(scope: Construct, id: string, props: ClusterRoleProps = {}) {
+    super(scope, id);
+
+    this._labelSelector = props.aggregationLabels ?? {};
+
+    this.apiObject = new k8s.KubeClusterRole(this, 'Resource', {
+      metadata: props.metadata,
+      rules: Lazy.any({ produce: () => this.synthesizeRules() }),
+      aggregationRule: Lazy.any({ produce: () => this.synthesizeAggregationRules() }),
+    });
+  }
+
+  /**
+   * Add the rules of the argument ClusterRole into this ClusterRole.
+   * @param role
+   */
+  public aggregateFrom(role: ClusterRole): void {
```

Comment:
I'm not sure about this API. Does an aggregation only makes sense with another managed role? What if I just need to add an aggregation label, pointing to an existing `ClusterRole`?

How about:

```ts
// aggregate rules from roles matching this label selector
public aggregate(key: string, value: string): void;

// combine rules from the given role by applying the appropriate aggregation labels
public combine(role: ClusterRole): void;
```

Reply:
Good idea 👍